src/db.rs

```diff
@@ -800,6 +862,57 @@ mod tests {
         }
     }
 
+    #[tokio::test]
```

Comment:
Here is one way option for adding tests.

- A set of, say 100 keys, 1-100
- Two tasks writing, one writing value to be same key, and other writing it to be key*2. 
- We wait for both to complete.
- Assert that either all values are same as key, or all values key*2.
- Repeat above loop few times.

Looking at the way locks are taken, it shouldn't be the case that there are partial writes. So, above is only for future regressions.

Reply:
Done: 1f0c3ed930426215ef2cec9693051e3df043e8e2